streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -373,8 +373,15 @@ public GroupAssignment assign(final Cluster metadata, final GroupSubscription gr
 
         final Set<TaskId> statefulTasks = new HashSet<>();
 
-        final boolean probingRebalanceNeeded =
-            assignTasksToClients(fullMetadata, allSourceTopics, topicGroups, clientMetadataMap, partitionsForTask, statefulTasks);
+        final boolean probingRebalanceNeeded;
+        try {
+            probingRebalanceNeeded = assignTasksToClients(fullMetadata, allSourceTopics, topicGroups, clientMetadataMap, partitionsForTask, statefulTasks);
+        } catch (final TaskAssignmentException | TimeoutException e) {
+            return new GroupAssignment(
```

Comment:
Since we could throw different exceptions here, would be good to add a log to indicate which type of exception is thrown.

Reply:
There are already corresponding log.error statement before those exceptions are thrown. No need to double log IMHO?

Reply:
Cool, sg.